react/wbooks/src/app/screens/SignUp/index.js

```diff
@@ -1,31 +1,43 @@
-import React, { useCallback } from 'react';
+import React, { useReducer, useCallback, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import i18next from 'i18next';
+import { Link } from 'react-router-dom';
 
 import FormWrapper from '../../components/Formik';
 import CustomField from '../../components/CustomField';
 import logoWolox from '../../assets/LogoWolox.png';
 import { validationSchemaSignUp } from '../../utils/validations';
 import { initialValuesSignUp } from '../../constants/auth';
+import { createUser, setToken } from '../../../services/AuthServices';
 
+import actionCreators from './reducer/actions';
 import styles from './styles.module.scss';
+import reducer, { initialState } from './reducer/reducer';
 
-const onSubmit = (values, actions) => {
+const onSubmit = values => {
+  localStorage.setItem('myData', values);
   setTimeout(() => {
     alert(JSON.stringify(values, null, 2));
-    actions.setSubmitting(false);
   }, 1000);
 };
 
 function SignUp() {
-  const onSubmitSignUp = useCallback((values, actions) => {
-    onSubmit(values, actions);
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  useEffect(() => {
+    createUser(initialValuesSignUp).then(response => {
+      setToken(response.data.access_token || '');
+      dispatch(actionCreators.createUser(response));
+    });
   }, []);
+
+  const onSubmitSignUp = useCallback(values => () => onSubmit(values), []);
```

Comment:
I think useCallback is not used correctly here. You are going to create an arrow function on every render. That will be the same as just doing 
`const onSubmitSignUp = () => onSubmit(values)`

I will suggest using the dependencies a little bit more, maybe like this
`const onSubmitSignUp = useCallback(() => onSubmit(state), [state]);`
And in the render you will use it directly like `onSubmitSignUp `

Do you see the difference?